### ISSUE TITLE
After upgrade to GO extension 0.6.56 version, some errors about package

### ISSUE THREAD

**Issue by a57571735** [Microsoft/vscode-go#892]
![gobug1](https://cloud.githubusercontent.com/assets/20694884/24490118/aecdff6e-1553-11e7-85e3-6da12a359d5c.png)
![gobug2](https://cloud.githubusercontent.com/assets/20694884/24490119/b02c6d5a-1553-11e7-9407-51846a35fbe6.png)

win7 64bit
vscode 1.10.2
how to fix it? thx

**Comment by a57571735** [comment]
![image](https://cloud.githubusercontent.com/assets/20694884/24493058/006e87ba-1560-11e7-9a28-a40463c62272.png)

**Comment by a57571735** [comment]
@mattetti @buckett @saml @AlekSi @neelance reinstall golang vscode and tools,this bug still here

**Comment by a57571735** [comment]
![image](https://cloud.githubusercontent.com/assets/20694884/24495070/951e45c0-1566-11e7-969a-1578d9738472.png)

**Comment by ansencumt** [comment]
![image](https://cloud.githubusercontent.com/assets/5181385/24496527/0fc40068-156b-11e7-9883-3cc860afe8dd.png)

![Uploading image.png…]()


I also get the same error.

**Comment by ansencumt** [comment]
The word "ode" is not exist in my system folder.

**Comment by a57571735** [comment]
I build 0.6.54, install it can work normal, remenber uninstall 0.6.56 first
Open the extensions menu in VSCode and choose Install from VSIX and select the Go-xxx.vsix
I upload zip file failed,  share url to download [Go-0.6.54.vsix](http://pan.baidu.com/s/1hsC8lpy)

**Comment by a57571735** [comment]
@ansencumt

**Comment by ansencumt** [comment]
@a57571735 thx

**Comment by nezorflame** [comment]
seems like it somehow trims the start of the package name thus failing to find the package.
@lukehoban please take a look.

**Comment by nezorflame** [comment]
Here's the previous build for the time being for those who like me are unable to install from previous link by @a57571735 
[Go-0.6.54.vsix.zip](https://github.com/Microsoft/vscode-go/files/881902/Go-0.6.54.vsix.zip)

**Comment by asdine** [comment]
I don't think it's related to Go 1.8 but only to the version `0.6.56`. The bug occured the moment i upgraded

**Comment by ramya-rao-a** [comment]
Hello people,

just saw this.
Thanks for reporting.

Most likely culprit is https://github.com/Microsoft/vscode-go/pull/864

Will send out a fix soon

**Comment by ramya-rao-a** [comment]
Looking at all the error messages posted here, it looks like a classic off by 1 error.
dataAccess -> ataAccess
code -> ode

But unfortunately, neither did I see this error while testing, nor can I see it now in any of my Go projects.

All of my Go project paths are of the form $GOPATH/src/github.com/somebody/someproject and I cant see this error in any of them.

So I created another project right under $GOPATH/src like @a57571735. Can't repro the bug then either :(

@a57571735 @nezorflame  Since you have local dev env available for the Go extension, can you help me debugging this issue? You need to put a breakpoint at https://github.com/Microsoft/vscode-go/blob/master/src/goCheck.ts#L167 and see what values you are getting for `currentGoWorkspace` and `importPath` variables

**Comment by nezorflame** [comment]
@ramya-rao-a sure, in a couple of minutes (on the road right now)

**Comment by ramya-rao-a** [comment]
@a57571735 @asdine @nezorflame @ansencumt
Can you share some details about your GOPATH?

- Is your GOPATH set to multiple workspaces
- Are you using the `go.gopath` setting to override the GOPATH env variable
- Do you have `go.inferGopath` set to `true` ?
- Can you run the command `Go: Current GOPATH` and see if that looks right to you?

**Comment by nezorflame** [comment]
- yeah (set in .bash_profile: GOPATH="${GOPATH1}:${GOPATH2}:${GOPATH3}")
- no
- no
- seems OK

**Comment by ramya-rao-a** [comment]
Alright, found the issue.

Commited the fix: https://github.com/Microsoft/vscode-go/commit/d417fd6725077d1233fb1bcd3aa5d097d02715a9

Will send out a release soon

**Comment by nezorflame** [comment]
@ramya-rao-a can confirm that it's fixed now (build from the latest comment).

**Comment by ramya-rao-a** [comment]
Thanks @nezorflame! The update is out as well.

**Comment by AaronReboot** [comment]
Hi, I was experiencing the issue described in this issue with 0.6.56. I just updated to 0.6.57 and there's still a problem:

It's now reporting "cannot import absolute path" at the top of all my package names

**Comment by ramya-rao-a** [comment]
@AaronReboot Can you share the build output from the output channel?

**Comment by ramya-rao-a** [comment]
@AaronReboot Make sure that the project you have opened is under the GOPATH that you get by running the command `Go: Current GOPATH`

**Comment by AaronReboot** [comment]
Sure thing. Here's my build output:
```Finished running tool: /Users/aaron/code/bin/golint

Finished running tool: /usr/local/go/bin/go build -i -o /var/folders/n_/7lfk4kw11hg0q83fy9kqc8jr0000gn/T/go-code-check -tags "" /Users/aaron/code/LLAPI
can't load package: package /Users/aaron/code/LLAPI: import "/Users/aaron/code/LLAPI": cannot import absolute path


Finished running tool: /usr/local/go/bin/go tool vet /Users/aaron/code/LLAPI/balance.go```

And here's my GOPATH
`Current GOPATH: /Users/aaron/code`

**Comment by ramya-rao-a** [comment]
Is there any reason, your project is directly under $GOPATH and not `$GOPATH/src`?

**Comment by AaronReboot** [comment]
It's just how things are organized at the firm I'm working for. It removes some of the complications with deploying App Engine projects.

I'd like to note there's been no difficulty with this simplified path structure prior to 0.6.56

**Comment by ramya-rao-a** [comment]
In 0.6.56 we pass the import path of the current package to `go build` 

And to find the import path we assume that the current file is under `$GOPATH/src`
If we can't find the import path this way, then the path to the current directory is used (which is wrong, causing the error you are seeing, we should have used `.`, we will fix that soon)

**Comment by liaoya** [comment]
I met the same problem on both my Linux and Windows VSCode 1.10.2 and Go 0.6.57. There is only one folder in my GOPATH. I also move the code to src and has the same problem.

The following is output on my Windows.
file: 'file:///d%3A/temp/gopl/src/helloworld/helloworld.go'
severity: 'Error'
message: 'can't load package: package d:/temp/gopl/src/helloworld: cannot find package "d:/temp/gopl/src/helloworld" in any of:
	C:\tools\go\src\d:\temp\gopl\src\helloworld (from $GOROOT)
	c:\opt\gopath\src\d:\temp\gopl\src\helloworld (from $GOPATH)
'
at: '1,1'
source: ''

**Comment by ramya-rao-a** [comment]
Since I am not able to repro, can you uninstall the Go extension you have, and install it using the [Go-0.6.58.vsix](https://github.com/Microsoft/vscode-go/blob/master/Go-0.6.58.vsix) to confirm if the fix works for you?

- Uninstall Go extension
- Download the vsix file [Go-0.6.58.vsix](https://github.com/Microsoft/vscode-go/blob/master/Go-0.6.58.vsix) 
- Run `code --install-extension Go-0.5.58.vsix` from the location where you downloaded the file

**Comment by ramya-rao-a** [comment]
@liaoya Please use the workaround above, that should fix the issue for you.

Also, I am curious about your set up. Do you have multiple GOPATHs set?
You are trying to build `d:/temp/gopl/src/helloworld` from which I am assuming `d:/temp/gopl` is your GOPATH. Can you run the command `Go: Current GOPATH` and share the result?

**Comment by liaoya** [comment]
The following is output of "go env" about gopath 
set GOPATH=c:\opt\gopath
I use 0.6.58 in issue #895, and it works as before.

I will not put the workspace in environment variable gopath and there's one item in gopath which store third party library and tool.

I'm just a beginner of Go. Thank you very much for the wonderful extension.

**Comment by AaronReboot** [comment]
Hi. I'm also happy to report that everything is fixed here too after manually installing 0.6.58!

**Comment by asdine** [comment]
Me too ! 👍

**Comment by lzelevinsky** [comment]
I had the same problem as others with 0.6.57. I have a single simple GOPATH.
Manually updating to 0.6.58 worked for me as well.

**Comment by gevg** [comment]
Me too ! :+1:

**Comment by ramya-rao-a** [comment]
I just released an update (0.6.59) yesterday which has the fix
